ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -454,7 +454,62 @@ public CompletableFuture<Short> undeleteBlobAsync(BlobId blobId, short lifeVersi
   @Override
   public short updateBlobExpiration(BlobId blobId, long expirationTime, CloudUpdateValidator cloudUpdateValidator)
       throws CloudStorageException {
-    return 0;
+    Timer.Context storageTimer = azureMetrics.blobUpdateTTLLatency.time();
+    AzureBlobLayoutStrategy.BlobLayout blobLayout = azureBlobLayoutStrategy.getDataBlobLayout(blobId);
+    String blobIdStr = blobLayout.blobFilePath;
+    BlobProperties blobProperties = getBlobProperties(blobLayout);
+    Map<String, String> cloudMetadata = blobProperties.getMetadata();
+
+    // Below is the correct behavior. For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
+    // We should never hit this case however because ReplicaThread::applyUpdatesToBlobInLocalStore does all checks.
+    if (cloudMetadata.containsKey(CloudBlobMetadata.FIELD_DELETION_TIME)
+        && Long.parseLong(cloudMetadata.get(CloudBlobMetadata.FIELD_DELETION_TIME)) != Utils.Infinite_Time) {
+      String error = String.format("Cannot update TTL of %s since it is already deleted in cloud.", blobIdStr);
+      throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.ID_Deleted));
+    }
+
+    /*
+    // Legacy cloudBlobStore does not expect an exception. However,
+    // below is the correct behavior. For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
+    // We should never hit this case however because ReplicaThread::applyUpdatesToBlobInLocalStore does all checks.
+    if (cloudMetadata.containsKey(CloudBlobMetadata.FIELD_EXPIRATION_TIME)
+        && Long.parseLong(cloudMetadata.get(CloudBlobMetadata.FIELD_EXPIRATION_TIME)) == Utils.Infinite_Time) {
+      String error = String.format("TTL of %s is already updated in cloud.", blobIdStr);
+      throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.Already_Updated));
+    }
+    */
```

Comment:
Minor: This block of code can be removed if it is intended to be commented.

Reply:
done